dbt/adapters/athena/connections.py

```diff
@@ -269,7 +269,7 @@ def process_query_stats(cursor: AthenaCursor) -> Tuple[int, int]:
         return cursor.rowcount, cursor.data_scanned_in_bytes
 
     def cancel(self, connection: Connection) -> None:
-        connection.handle.cancel()
+        pass
```

Comment:
could you confirm this will not impact on `KeyboardInterrupt` case ? It's crucial for me when I run dbt jobs via airflow.

Reply:
I executed `dbt run` command using my branch and dbt-athena1.6.0 on the CLI, and sending Ctrl+C intermediately.
Stacktraces in logs change as follows, but the processing itself completes at almost the same speed.

#### Part of stdout from my branch

Model names and virtualenv's name are edited for masking internal information of the company I belonging to. 

```
^C23:25:06  2 of 15 OK created sql view model model1 ............ [OK -1 in 1.89s]
23:25:06  CANCEL query model.model2 ......................... [CANCEL]
23:25:06  CANCEL query model.model1 ........................... [CANCEL]
23:25:06  1 of 15 OK created sql view model model1 .......... [OK -1 in 1.92s]
23:25:06
23:25:06  Exited because of keyboard interrupt
23:25:06
23:25:06  Done. PASS=0 WARN=0 ERROR=0 SKIP=0 TOTAL=0
23:25:06
23:25:06  Finished running  in 0 hours 0 minutes and 9.90 seconds (9.90s).
23:25:06  Encountered an error:

23:25:06  Traceback (most recent call last):
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 87, in wrapper
    result, success = func(*args, **kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 72, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 143, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 172, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 219, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 259, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/main.py", line 579, in run
    results = task.run()
              ^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 469, in run
    result = self.execute_with_hooks(selected_uids)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 430, in execute_with_hooks
    res = self.execute_nodes()
          ^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 356, in execute_nodes
    self.run_queue(pool)
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 269, in run_queue
    node = self.job_queue.get()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/graph/queue.py", line 125, in get
    _, node_id = self.inner.get(block=block, timeout=timeout)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.pyenv/versions/3.11.2/lib/python3.11/queue.py", line 171, in get
    self.not_empty.wait()
  File "/home/horiuchi/.pyenv/versions/3.11.2/lib/python3.11/threading.py", line 320, in wait
    waiter.acquire()
KeyboardInterrupt
```

#### Part of stdout from dbt-athena1.6.0

```
^C23:28:47
23:28:47  Finished running 1 view model in 0 hours 0 minutes and 7.54 seconds (7.54s).
23:28:47  Encountered an error:
'Connection' object has no attribute 'cancel'
23:28:47  Traceback (most recent call last):
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 356, in execute_nodes
    self.run_queue(pool)
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 269, in run_queue
    node = self.job_queue.get()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/graph/queue.py", line 125, in get
    _, node_id = self.inner.get(block=block, timeout=timeout)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.pyenv/versions/3.11.2/lib/python3.11/queue.py", line 171, in get
    self.not_empty.wait()
  File "/home/horiuchi/.pyenv/versions/3.11.2/lib/python3.11/threading.py", line 320, in wait
    waiter.acquire()
KeyboardInterrupt

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 87, in wrapper
    result, success = func(*args, **kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 72, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 143, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 172, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 219, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/requires.py", line 259, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/cli/main.py", line 579, in run
    results = task.run()
              ^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 469, in run
    result = self.execute_with_hooks(selected_uids)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 430, in execute_with_hooks
    res = self.execute_nodes()
          ^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 381, in execute_nodes
    self._cancel_connections(pool)
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/task/runnable.py", line 329, in _cancel_connections
    for conn_name in adapter.cancel_open_connections():
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/adapters/base/impl.py", line 1109, in cancel_open_connections
    return self.connections.cancel_open()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/adapters/sql/connections.py", line 43, in cancel_open
    self.cancel(connection)
  File "/home/horiuchi/.local/share/virtualenvs/mT38Pn63/lib/python3.11/site-packages/dbt/adapters/athena/connections.py", line 272, in cancel
    connection.handle.cancel()
    ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Connection' object has no attribute 'cancel'
```

Reply:
Should be fine I believe - the concept of "connection" with athena is not really existing, as under the hood pyathena make boto3 calls.

Reply:
The question is following: once you send SIGINT to the dbt process, will it also cancel Athena query ? You may check it with `dbt --debug run ...` or directly in AWS Athena console.

Reply:
@svdimchenko I got now - and make sense, e.g. if a task is stopped in airflow, you want to stop the queries in athena.

@keshihoriuchi could you have a look if we have this behavior?

Reply:
I tried running `dbt --debug`, but it turned out as follows. Queries are not canceled.
```
^C08:56:56  CANCEL query model.model1  [CANCEL]
08:56:56  CANCEL query model.model2 ........................ [CANCEL]
08:56:57  SQL status: OK -1 in 2.0 seconds
08:56:57  Timing info for model.model2 (execute): 17:56:55.682531 => 17:56:57.245977
08:56:57  On model.model2: Close
08:56:57  3 of 15 OK created sql view model model2 ......... [OK -1 in 1.57s]
08:56:57  Finished running node model.model2
08:56:57  SQL status: OK -1 in 2.0 seconds
08:56:57  Timing info for model.model1 (execute): 17:56:55.822809 => 17:56:57.473540
08:56:57  On model.model1: Close
08:56:57  4 of 15 OK created sql view model model1  [OK -1 in 1.67s]
08:56:57  Finished running node model.model1
08:56:57
08:56:57  Exited because of keyboard interrupt
08:56:57
08:56:57  Done. PASS=2 WARN=0 ERROR=0 SKIP=0 TOTAL=2
```

Probably because when `materialized` is set to "view," the execution ends immediately, so I couldn't get the expected result. Next, I'll try changing it to "table."

And from what I understood by reading the code, query cancellation seems to be happening on AthenaCursor https://github.com/dbt-athena/dbt-athena/blob/main/dbt/adapters/athena/connections.py#L109 .